src/gateways/BeaconsApiAccountHolderGateway.ts

```diff
@@ -12,22 +14,21 @@ export class BeaconsApiAccountHolderGateway implements AccountHolderGateway {
   private readonly accountHolderControllerRoute = "account-holder";
   private readonly accountHolderIdEndpoint = "auth-id";
   private readonly accountHolderBeaconsEndpoint = "beacons";
+  private readonly authGateway: AuthGateway;
 
-  constructor(apiUrl: string) {
+  constructor(apiUrl: string, authGateway: AuthGateway = new AadAuthGateway()) {
```

Comment:
While working out how to have Cypress POST a DraftRegistration to the Service, I kept wondering why AccountHolderGateway's clients need to provide it with an access token.  I was writing code that would run inside Cypress that would get an access token to allow AccountHolderGateway to talk to the Service.  Why does Cypress (or another client) need to do this step?

This felt like a leaky abstraction to me.  I _think_ domain code should just be able to go to an AccountHolderGateway and say "get me an AccountHolder"?  It shouldn't have to say "here's a string so you can talk to your API, now go get me an AccountHolder".

So I tried out having AccountHolderGateway fully encapsulate communication with the API that serves AccountHolders.

Here, the BeaconsApiAccountHolderGateway has a default AuthGateway, the concrete AadAuthGateway.  The BeaconsApiAccountHolderGateway handles API-level concerns like getting an access token.  The AuthGateway can be swapped out on instantiation for ease of testing, e.g. in the Cypress code above which uses a AadAuthGateway with a different config.

I chose to use the default parameter method of dependency injection rather than putting `AadAuthGateway` on the application container so that we're not tempted to create leaky abstractions by using `AadAuthGateway` directly in the future.  Because the `AuthGateway` is relevant only to how the Webapp communicates with the Service, and because we encapsulated that communication in Gateways, an `AuthGateway` should only be used inside a Gateway.

Thoughts?

Reply:
Changed approach slightly to inject the AuthGateway inside the appcontainer rather than as a default parameter inside the constructor of the entity gateways.